### ISSUE TITLE
[HOLD for Payment 2024-09-09][$250] Wrong parsing when single backticks used in 2 separate places

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46917]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.17-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @coleaeason 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1722625457122619

## Action Performed:
1. Open any report
2. Send the string  "it uses `;` separated fields because the line number column could have `:` in them"
## Expected Result:
it uses` ; `separated fields because the line number column could have` : `in them
## Actual Result:
it uses `;` separated fields because the line number column could have :` in them`
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![Screenshot 2024-08-02 at 3 03 21 PM](https://github.com/user-attachments/assets/ca58f88b-6c97-4b77-923c-02a6d27331aa)

![Snip - (13) New Expensify - Google Chrome (2)](https://github.com/user-attachments/assets/3430ef46-1936-4e70-86b1-b0af49e009e6)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014560c8f3601ebcb2</li>
        <li>Upwork Job ID: 1823820445986093493</li>
        <li>Last Price Increase: 2024-08-14</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 103588545</li>
<li>tsa321 | Contributor | 103588546</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ikevin127</details>

**Comment by JmillsExpensify** [comment]
Hmm @thienlnam should we resolve this issue as part of on-going maintenance of our markdown library?

**Comment by thienlnam** [comment]
We've been making some of these external - cc @BartoszGrajdek Looks like this one can also be external right? Or would you like to take it

**Comment by BartoszGrajdek** [comment]
> Looks like this one can also be external right?

Yes, this is something External contributors can definitely work on 🙌🏻

**Comment by ikevin127** [comment]
♻️ Reviewing proposal today.

**Comment by ikevin127** [comment]
@tsa321 Thanks for the proposal, noticed quite a few edits there 😅 Unfortunately even though the regex you suggested seems to fix the issue according to the issue's **Expected result**, it fails the following **:x: 6 tests** in
`ExpensiMark-HTML-test.js`:

- :x: **Test inline code blocks with triple backticks**
- :x: **Test inline code with one backtick as content**
- :x: **Test inline code with multiple backtick symbols as content**
- :x: **Test for codefence with no content**
- :x: **Video markdown conversion to html tag › inline code in alt**
- :x: **Video markdown conversion to html tag › blockquote in alt**

##

Besides the failing tests, which make the proposed solution **unacceptable**, I also noticed that given the following input:

```it uses`;`separated fields because the line number column could have`:`in them ` ` with space```

Once sent, the parsing will have the message look like this:

<img width="659" alt="Screenshot 2024-08-15 at 15 43 02" src="https://github.com/user-attachments/assets/b4f90fe2-10d7-4558-843b-48279693b79a">

causing white-space (empty space) inline code-block to not be parsed and instead showing the back-ticks instead of an empty code-block.

##

**If still interested in fixing the issue I suggest cloning the [`expensify-common`](https://github.com/Expensify/expensify-common) repository and running the tests as you iterate and try to come-up with a regex that would fix our issue while not breaking any of the tests (without adjusting the tests to "match" the new regex).**

**Comment by ikevin127** [comment]
@tsa321's [updated proposal](https://github.com/Expensify/App/issues/46917#issuecomment-2272779877) looks good to me, all **expensify-common** tests are 🟢 **Passing**. The RCA is correct and the proposed solution fixes the issue as per the **Expected result**.

🎀👀🎀 C+ reviewed

**Comment by ikevin127** [comment]
## 🗒️ Some things to note

> [!important]
> These are out of scope for our current issue since they are present on current **staging** as well.

**1. White-space (empty space) inline code-block showing the back-ticks instead of an empty code-block**

Given the following input:

```it uses`;`separated fields because the line number column could have`:`in them ` ` with space```

Once sent, the parsing will have the message look like this:

<img width="659" alt="Screenshot 2024-08-15 at 15 43 02" src="https://github.com/user-attachments/assets/b4f90fe2-10d7-4558-843b-48279693b79a">

causing white-space (empty space) inline code-block to not be parsed and instead showing the back-ticks instead of an empty code-block.

**2. Nested inline code-block parsing is broken (compared to GitHub's for example)**

Given the following input:

```
```nest 1 `nest 2` nest 1```
```

Once sent, the parsing will have the message look like this:

<img width="302" alt="Screenshot 2024-08-16 at 14 12 09" src="https://github.com/user-attachments/assets/bb8c6722-07f9-4439-83fc-ef65246f45f5">

For reference this is how GitHub is parsing it:

<img width="268" alt="Screenshot 2024-08-16 at 14 12 19" src="https://github.com/user-attachments/assets/1f22bc71-68ae-4e46-a2ea-5babf63431c9">

cc @BartoszGrajdek

**Comment by dangrous** [comment]
Do we want to open separate issues for those two bugs you noted? Not sure if they exist already.

For this one, the proposal sounds good!

**Comment by ikevin127** [comment]
Reviewing PR today!

**Comment by ikevin127** [comment]
🟢 `expensify-common` PR was reviewed, more details in https://github.com/Expensify/expensify-common/pull/785#issuecomment-2299909099. 🔖 Will complete the PR Reviewer Checklist on the E/App PR once the `expensify-common` PR is merged.

**Comment by dangrous** [comment]
e-c PR is merged!

**Comment by BartoszGrajdek** [comment]
@ikevin127 sorry for the late response to [this comment](https://github.com/Expensify/App/issues/46917#issuecomment-2294328303) - it got buried under too many GH notifications 😅 

I agree with the 1st note, this is probably something we could think about improving.

As for the 2nd note this is actually something we had a lengthy discussion about [right here](https://swmansion.slack.com/archives/C06BDSWLDPB/p1713365698298649?thread_ts=1713274778.358479&cid=C06BDSWLDPB) - let me know if you have access to it. There's no support for an "_inline_" code block (triple backticks) if you wish to create a code block linebreaks are required between backticks and the content itself.

**Comment by dangrous** [comment]
Thanks for the input @BartoszGrajdek!

Do you (both) think it's worth it for me to create an issue to look into the first note and fix it? Seems pretty minor but I guess it can't hurt

**Comment by ikevin127** [comment]
⚠️ Production deploy automation failed here -> this should be on **[HOLD for Payment 2024-09-09]** according to yesterday's production deploy from [comment](https://github.com/Expensify/App/pull/47838#issuecomment-2325378038).

cc @dangrous @JmillsExpensify

**Comment by ikevin127** [comment]
cc @JmillsExpensify

**Comment by ikevin127** [comment]
cc @JmillsExpensify bump for payment

**Comment by ikevin127** [comment]
cc @JmillsExpensify bump for payment

**Comment by JmillsExpensify** [comment]
Payment summary:

 - Contributor: @tsa321 $250
 - Reviewer: @ikevin127 $250

**Comment by JmillsExpensify** [comment]
@ikevin127 do we have a BZ checklist for this issue yet?

**Comment by ikevin127** [comment]
- Not sure whether regression test is needed for this one because on expensify-common PR https://github.com/Expensify/expensify-common/pull/785 we added unit tests, but anyway.

##

* [x]  [@ikevin127] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/expensify-common/pull/690
* [x]  [@ikevin127] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/expensify-common/pull/690/files#r1757521200
* [x]  [@ikevin127] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**.
* [x]  [@ikevin127] Determine if we should create a regression test for this bug.
* [x]  [@ikevin127] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal

1. Open any report.
2. Send the following message: 
```
test `;` test `:` test
```
4. Verify that the message is displayed correctly as:

    test `;` test `:` test

##

cc @JmillsExpensify 

Do we agree 👍 or 👎.

**Comment by JmillsExpensify** [comment]
Contributors paid via Upwork and regression test created. I'm closing this one.
